### PR TITLE
Add add_slice_to_data

### PR DESCRIPTION
**Pull request by wthrowe**

### CODE REVIEW

File: src/DataStructures/VariablesHelpers.hpp [hunk 1]
```diff
@@ -43,3 +46,32 @@ Variables<TagsList> data_on_slice(const Variables<TagsList>& vars,
   }
   return interface_vars;
 }
+
+/*!
+ * \ingroup DataStructures
+ * \brief Adds data on a codimension 1 slice `slice` to a volume
+ * quantity `data`. The slice has a constant grid coordinate in
```
Comment (osheamonn) [inline src/DataStructures/VariablesHelpers.hpp]: Is it not the case that the slice has constant logical coordinate and not grid coordinate?

File: src/DataStructures/VariablesHelpers.hpp [hunk 2]
```diff
@@ -11,13 +11,16 @@
 #include "DataStructures/Index.hpp"
 #include "DataStructures/SliceIterator.hpp"
 #include "DataStructures/Variables.hpp"
+#include "Utilities/Gsl.hpp"
 
 /*!
  * \ingroup DataStructures
  * \brief Slices the data within `vars` to a codimension 1 slice. The
  * slice has a constant grid coordinate in direction `sliced_dim`, and slices
```
Comment (kidder) [inline src/DataStructures/VariablesHelpers.hpp]: this should be logical coordinate as well

**Review by kidder — CHANGES_REQUESTED** [review]

File: src/DataStructures/VariablesHelpers.hpp [hunk 3]
```diff
@@ -43,3 +47,33 @@ Variables<TagsList> data_on_slice(const Variables<TagsList>& vars,
...
+ * \brief Adds data on a codimension 1 slice `slice` to a volume
+ * quantity `data`. The slice has a constant logical coordinate in
+ * direction `sliced_dim`, and slices the volume through the point
+ * `slice_point` of this coordinate.
+ *
+ * \see data_on_slice
+ */
+template <std::size_t VolumeDim, typename TagsList>
+void add_slice_to_data(const gsl::not_null<Variables<TagsList>*> data,
+                       const Variables<TagsList>& slice,
```
Comment (kidder) [inline src/DataStructures/VariablesHelpers.hpp]: slice -> vars_on_slice

File: src/DataStructures/VariablesHelpers.hpp [hunk 4]
```diff
@@ -43,3 +47,33 @@ Variables<TagsList> data_on_slice(const Variables<TagsList>& vars,
...
+ * \ingroup DataStructures
+ * \brief Adds data on a codimension 1 slice `slice` to a volume
+ * quantity `data`. The slice has a constant logical coordinate in
+ * direction `sliced_dim`, and slices the volume through the point
+ * `slice_point` of this coordinate.
+ *
+ * \see data_on_slice
+ */
+template <std::size_t VolumeDim, typename TagsList>
+void add_slice_to_data(const gsl::not_null<Variables<TagsList>*> data,
```
Comment (kidder) [inline src/DataStructures/VariablesHelpers.hpp]: volume_vars

File: src/DataStructures/VariablesHelpers.hpp [hunk 5]
```diff
@@ -43,3 +47,33 @@ Variables<TagsList> data_on_slice(const Variables<TagsList>& vars,
...
+                       const Variables<TagsList>& slice,
+                       const Index<VolumeDim>& extents,
+                       const size_t sliced_dim,
+                       const size_t slice_point) {
+  const size_t number_of_independent_components =
```
Comment (kidder) [inline src/DataStructures/VariablesHelpers.hpp]: get constexpr

**Review by kidder — CHANGES_REQUESTED** [review]

File: src/DataStructures/VariablesHelpers.hpp [hunk 6]
```diff
@@ -11,12 +11,16 @@
...
 #include "DataStructures/Variables.hpp"
+#include "Utilities/Gsl.hpp"
 
 /*!
  * \ingroup DataStructures
  * \brief Slices the data within `vars` to a codimension 1 slice. The
- * slice has a constant grid coordinate in direction `sliced_dim`, and slices
- * the volume through the point `slice_point` of this coordinate.
+ * slice has a constant logical coordinate in direction `sliced_dim`,
+ * and slices the volume through the point `slice_point` of this
```
Comment (kidder) [inline src/DataStructures/VariablesHelpers.hpp]: I find "the point `slice_point` of this coordinate" to be confusing as you are not supposed to pass the value of the fixed coordinate, but instead the index that is fixed in `sliced_dim`.  

How about something like:
 "and slices the volume at `fixed_index` in `sliced_dim`.  For example, to get the lower boundary of `sliced_dim`,  pass `0` for `fixed_index`; to get the upper boundary, pass `extents[sliced_dim] - 1`"

Thoughts?

File: src/DataStructures/VariablesHelpers.hpp [hunk 7]
```diff
@@ -43,3 +47,33 @@ Variables<TagsList> data_on_slice(const Variables<TagsList>& vars,
   }
   return interface_vars;
 }
+
+/*!
+ * \ingroup DataStructures
+ * \brief Adds data on a codimension 1 slice to a volume quantity. The
+ * slice has a constant logical coordinate in direction `sliced_dim`,
+ * and slices the volume through the point `slice_point` of this
```
Comment (kidder) [inline src/DataStructures/VariablesHelpers.hpp]: see comment above

**Review by nilsdeppe — DISMISSED** [review]
squash

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**